### ISSUE TITLE
err: improve `Multi-output expressions are not supported in this context` error message

### ISSUE THREAD

**Issue by MarcoGorelli** [narwhals-dev/narwhals#920]
example of how to get it:
```python
In [1]: import narwhals as nw

In [2]: import pandas as pd

In [3]: nw.from_native(pd.Series([1,2,3]), series_only=True) == [1,2,3]
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
Cell In[3], line 1
----> 1 nw.from_native(pd.Series([1,2,3]), series_only=True) == [1,2,3]

File ~/scratch/.venv/lib/python3.12/site-packages/narwhals/series.py:1487, in Series.__eq__(self, other)
   1485 def __eq__(self, other: object) -> Self:  # type: ignore[override]
   1486     return self._from_compliant_series(
-> 1487         self._compliant_series.__eq__(self._extract_native(other))
   1488     )

File ~/scratch/.venv/lib/python3.12/site-packages/narwhals/_pandas_like/series.py:238, in PandasLikeSeries.__eq__(self, other)
    236 def __eq__(self, other: object) -> PandasLikeSeries:  # type: ignore[override]
    237     ser = self._native_series
--> 238     other = validate_column_comparand(self._native_series.index, other)
    239     return self._from_native_series(self._rename(ser.__eq__(other), ser.name))

File ~/scratch/.venv/lib/python3.12/site-packages/narwhals/_pandas_like/utils.py:40, in validate_column_comparand(index, other)
     37     if len(other) > 1:
     38         # e.g. `plx.all() + plx.all()`
     39         msg = "Multi-output expressions are not supported in this context"
---> 40         raise ValueError(msg)
     41     other = other[0]
     42 if isinstance(other, PandasLikeDataFrame):

ValueError: Multi-output expressions are not supported in this context
```

**Comment by FBruzzesi** [comment]
Should this comparison be supported? Polars allows it if the comparand is of the same length, and panics otherwise.
I am opening a PR for how it would look like 😇

**Comment by DeaMariaLeon** [comment]
@MarcoGorelli do we still want this?

**Comment by MarcoGorelli** [comment]
If it's possible to give a good message, then yes 👍

**Comment by MarcoGorelli** [comment]
closed by #1382
